h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -78,6 +78,13 @@ public void stop() {
         if(null != this.trackerThread) {
             this.trackerThread.interrupt();
             this.trackerThread = null;
+            try {
+                this.sock.close();
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to close Rabit tracker socket.", e);
+            } finally {
+                this.port = 9091;
```

Comment:
Why does the port need to be returned to the default? Would it be cleaner to invalidate this object and make the caller create a new instance of the tracker?

Reply:
During start I'm only checking ports until `this.port > 9999` (can change that, though, just did it because the original tracker was) so thought that resetting the port on stop is a good idea.

Can invalidate the object, but that would require probably an additional flag, saying the object got stopped and some easy logic in start so probably similar complexity as resetting in stop?

Reply:
yes, that is true - it just seems safer - the one flag will never change, however people can add more stuff to lets say start and forget about handling the reset of those values